docs/how-to-guides/develop-a-tool/create-your-own-custom-strong-type-connection.md

```diff
@@ -0,0 +1,107 @@
+# Create and Use Your Own Custom Strong Type Connection
+Connections serve as a secure method for managing credentials for external APIs and data sources. This document provides a step-by-step guide on how to create and use a custom strong type connection. The advantages of using a custom strong type connection are as follows:
```

Comment:
Connections provide a secure method for managing credentials for external APIs and data sources in Prompt flow. This guide explains how to create and use a custom strong type connection.

## What is a Custom Strong Type Connection?
A custom strong type connection in Prompt flow allows you to define a custom connection class with strongly typed keys. This provides the following benefits:

- Enhanced user experience - no need to manually enter connection keys.
- Rich intellisense experience - defining key types enables real-time suggestions and auto-completion of available keys as you work in VS Code.
- Central location to view available keys and data types.